cmd/updater/update.sh

```diff
@@ -330,14 +330,27 @@ function validate_update() {
     return 0
 }
 
+function run_systemd_action() {
+    local action=$1
+    local data_dir=$2
+
+    # Get the owner of a running process.
+    owner=$(awk '{ print $1 }' <(ps aux | grep "[a]lgod"))
+    if [[ "$owner" = "root" ]] || grep sudo <(groups "$owner"); then
+        sudo -n systemctl "${action}" "algorand@$(systemd-escape "${data_dir}")"
```

Comment:
I think that before you're going to do that, you need to verify that you're running as root. If not, this operation is going to fail anyway.
( and it need to be checked at the very beginning of the update script ).